docs/source/project_documentation/dev_guide/plugin_dev_guide.md

```diff
@@ -4,83 +4,309 @@
+
+Example project model:
+```yaml
+project:
+  name: My project
+  description: |
+    An example AaC project definition.
+```
+
+You can now use the `gen-project` command to create the necessary project structure.
+
+```bash
+aac gen-project ./my_project.aac --output .
+```
+
+If you ever need a quick refresher on the gen-project command and it's options you can run:
+```bash
+$ aac gen-project -h
+Usage: aac gen-project [OPTIONS] AAC_PROJECT_FILE
+
+Options:
+  --output TEXT      The location to output generated plugin code.
+  --no-prompt        Informs gen-plugin to execute without asking the user to
+                     confirm output paths.
+  --force-overwrite  Informs generator to backup and overwrite all existing
+                     files regardless of template definition.
+  --evaluate         Informs generator to only write evaluation files with no
+                     impact to existing files.
+  -h, --help         Show this message and exit.
+```
+
+This will create a project structure that looks like this:
+
+```markdown
+├── README.md
+├── setup.py
+├── docs
+├── src
+├── tests
+└── tox.ini
+```
+
 ## Generating AaC Plugins (Gen-Plugin Command)
 
-AaC comes pre-packaged with a plugin called `Gen-Plugin` which will take an AaC architecture file describing a plugin and then generate the necessary code stubs and python project infrastructure for the user to begin implementing their new plugin.
+AaC comes pre-packaged with a plugin called `Gen-Plugin` which will take an AaC architecture file describing a plugin and then generate the necessary code stubs project infrastructure for the user to begin implementing their new plugin.
 
 ### Gen-Plugin Usage and Output
 
-The `gen-plugin` command can be executed as follows: `aac gen-plugin ./path/to/architecture_file.yaml`
+The `gen-plugin` command can be executed as follows:
+
+```bash
+$ aac gen-plugin -h
+Usage: aac gen-plugin [OPTIONS] AAC_PLUGIN_FILE
+
+Options:
+  --code-output TEXT  The location to output generated plugin code.
+  --test-output TEXT  The location to output generated plugin test code.
+  --doc-output TEXT   The location to output generated plugin documentation
+                      code.
+  --no-prompt         Informs gen-plugin to execute without asking the user to
+                      confirm output paths.
+  --force-overwrite   Informs generator to backup and overwrite all existing
+                      files regardless of template definition.
+  --evaluate          Informs generator to only write evaluation files with no
+                      impact to existing files.
+  -h, --help          Show this message and exit.
+```
+
+While the `gen-plugin` command will attempt to sort out directory paths for you if not provided, it's often simplest to just provide `--code-output` and `--test-output` to ensure the generated code is placed where you want it.  Currently the `gen-plugin` command doesn't output documentation, but the option is in place for future use.
```

Comment:
A readme.md file is created when generating projects but not plugins...but that really isn't what was intended.  In this instance I was thinking about potentially generating something like the content in `/docs/source/project_documentation/plugins` rather than a readme.md file.  I like the idea of doing both in the future and recommend we create a story to look into it.